go/vt/vttablet/tabletserver/vstreamer/helper_event_test.go

```diff
@@ -0,0 +1,569 @@
+					}
+					if !ts.fieldEventsSent[table] {
+						output = append(output, fe.String())
+						ts.fieldEventsSent[table] = true
+					}
+					output = append(output, ts.getRowEvent(table, bv, fe, stmt, uint32(flags)))
+				}
+			}
+
+		}
+		tc.input = input
+		tc.output = append(tc.output, output)
+		testcases = append(testcases, tc)
+	}
+	runCases(ts.t, ts.options.filter, testcases, "current", nil)
+}
+
+func (ts *TestSpec) getFieldEvent(table *schemadiff.CreateTableEntity) *TestFieldEvent {
+	var tfe TestFieldEvent
+	tfe.table = table.Name()
+	tfe.db = testenv.TestDBName
+	for _, col := range table.TableSpec.Columns {
+		tc := TestColumn{}
+		tc.name = col.Name.String()
+		sqlType := col.Type.SQLType()
+		tc.dataType = sqlType.String()
+		tc.dataTypeLowered = strings.ToLower(tc.dataType)
+		tc.collate = col.Type.Options.Collate
+		switch tc.dataTypeLowered {
+		case "int32":
+			tc.len = 11
+			tc.charset = 63
+			tc.colType = "int(11)"
+		case "varchar", "varbinary", "char", "binary":
+			l, _ := strconv.Atoi(col.Type.Length.Val)
+			switch tc.dataTypeLowered {
+			case "binary", "varbinary":
+				tc.len = int64(l)
+				tc.charset = 63
+			default:
+				tc.len = 4 * int64(l)
+				tc.charset = 45
+				if tc.dataTypeLowered == "char" && strings.Contains(tc.collate, "bin") {
+					tc.dataType = "BINARY"
+				}
+			}
+			tc.colType = fmt.Sprintf("%s(%d)", tc.dataTypeLowered, l)
+		case "blob":
+			tc.len = 65535
+			tc.charset = 63
+			tc.colType = "blob"
+		case "text":
+			tc.len = 262140
+			tc.charset = 45
+			tc.colType = "text"
+		case "set":
+			tc.len = 56
+			tc.charset = 45
+			tc.colType = fmt.Sprintf("%s(%s)", tc.dataTypeLowered, strings.Join(col.Type.EnumValues, ","))
+			ts.metadata[getMetadataKey(table.Name(), tc.name)] = col.Type.EnumValues
+		case "enum":
+			tc.len = int64(len(col.Type.EnumValues) + 1)
+			tc.charset = 45
+			tc.colType = fmt.Sprintf("%s(%s)", tc.dataTypeLowered, strings.Join(col.Type.EnumValues, ","))
+			ts.metadata[getMetadataKey(table.Name(), tc.name)] = col.Type.EnumValues
+		default:
+			log.Infof(fmt.Sprintf("unknown sqlTypeString %s", tc.dataTypeLowered))
+		}
+		tfe.cols = append(tfe.cols, &tc)
+	}
+	return &tfe
+}
+
+func getMetadataKey(table, col string) string {
+	return fmt.Sprintf("%s:%s", table, col)
+}
+
+func (ts *TestSpec) setMetadataMap(table, col, value string) {
+	values := strings.Split(value, ",")
+	valuesReversed := make([]string, len(values))
+	for i, v := range values {
+		valuesReversed[len(values)-1-i] = v
+	}
+	ts.metadata[getMetadataKey(table, col)] = valuesReversed
+}
+
+func (ts *TestSpec) getMetadataMap(table string, col *TestColumn, value string) string {
+	var bits int64
+	value = strings.Trim(value, "'")
+	meta := ts.metadata[getMetadataKey(table, col.name)]
+	values := strings.Split(value, ",")
+	for _, v := range values {
+		v2 := strings.Trim(v, "'")
+		for i, m := range meta {
+			m2 := strings.Trim(m, "'")
```

Comment:
It seems like all of these Trims assume that a char based types would have values quoted using `'`? Does this mean that the tests will break if we use `"`? Is this clear to the writer anywhere? Maybe it is.

Reply:
i second this question

Reply:
Yes, this is an assumption. Added the related assumptions to the comments at the top of the file, so it is clearer. These are tests in our control. We can change it in a future PR if we find the need for it.